### PR TITLE
User mentions from Slack to Discord

### PR DESCRIPTION
**Pull request by Dante383**
#7 for Slack to Discord

### CODE REVIEW

File: src/Slack.py [hunk 1]
```diff
@@ -20,12 +20,12 @@
...
 handler.setLevel(logging.INFO)
 
-formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
 handler.setFormatter(formatter)
 
 logger.addHandler(handler)
 
-# Check if Api Key are supplied. Else the bridge won't be able to start at all.
+# Check if Api Key are supplied. Else the bridge won"t be able to start at all.
```
Comment (MatejMecka) [inline src/Slack.py]: it's won't not won"t

**Review by MatejMecka — CHANGES_REQUESTED** [review]

File: src/Slack.py [hunk 2]
```diff
@@ -38,13 +38,25 @@
 sc = SlackClient(SlackApiKey)
 sc.rtm_connect()
 
-# Set up URL's and headers for Discord.
+# Set up URL"s and headers for Discord.
```
Comment (MatejMecka) [inline src/Slack.py]: URL's not URL"s

File: src/Slack.py [hunk 3]
```diff
@@ -38,13 +38,25 @@
...
 headers = { "Authorization":"Bot {}".format(DiscordApiKey),
 			"User-Agent":"DiscordSlackBridge (http://reddit.com/r/teendeveloper, v0.1)",
 "Content-Type":"application/json", }
 
 # Run the Bridge 
 
+# we have to query all users in slack to map their id to their names later. querying every message
```
Comment (MatejMecka) [inline src/Slack.py]: Uppercase We, Uppercase Querying

File: src/Slack.py [hunk 4]
```diff
@@ -54,23 +66,48 @@
...
 			discordChannel = ""
+
 			if channel in channels:
 				discordChannel = channels.get(channel)
 			else:
 				pass	
+
+			# message formatting section
+
+			# mentions. valid mention in slack looks like this: <@username>
```
Comment (MatejMecka) [inline src/Slack.py]: Mentions capitalized

File: src/Slack.py [hunk 5]
```diff
@@ -54,23 +66,48 @@
...
+
 			# Handle Channels
 			discordChannel = ""
+
 			if channel in channels:
 				discordChannel = channels.get(channel)
 			else:
 				pass	
+
+			# message formatting section
```
Comment (MatejMecka) [inline src/Slack.py]: Message capitalized

File: src/Slack.py [hunk 6]
```diff
@@ -54,23 +66,48 @@
...
+
+			# mentions. valid mention in slack looks like this: <@username>
+			if "@" in message: 
+				# split message to a spaces and then detect which of the expressions starts with <@ and ends with >
```
Comment (MatejMecka) [inline src/Slack.py]: Split capitalized

File: src/Slack.py [hunk 7]
```diff
@@ -54,23 +66,48 @@
...
+				expressions = message.split(' ')
+				for expression in expressions:
+					if expression[:2] == '<@' and expression[-1:] == '>':
+						userId = expression[2:-1]
+						# now lets map it to username (if usersList exists)
```
Comment (MatejMecka) [inline src/Slack.py]: Now capitalized

File: src/Slack.py (hunk 2 above)
Comment (Dante383) [reply src/Slack.py]: yeah autoreplacing, also we should change " to ' in Discord.py too

**Review by MatejMecka — CHANGES_REQUESTED** [review]
Just Capitalize the sentences. If you see the other comments they start with an uppercase letter. Feels Nicer and easier to read

File: src/Slack.py (hunk 6 above)
Comment (MatejMecka) [reply src/Slack.py]: Mentions too

File: src/Slack.py (hunk 2 above)
Comment (MatejMecka) [reply src/Slack.py]: For comments no. But Sure.

**Review by MatejMecka — APPROVED** [review]

**Pull request merged**